proxy/reverse.go

```diff
@@ -60,6 +60,10 @@ func removeSingleHopHeaders(hdrs *http.Header) {
 type reverseProxy struct {
 	director  *director
 	transport http.RoundTripper
+
+	// proxyID maps to the endpoints of the proxy.
+	proxyID   string
+	endpoints map[string]struct{}
```

Comment:
I don't we need to pass the endpoints to http header either. We can only pass the `proxyID` and detect by ID, and once we detect it, we map to `endpoints` from this struct and ban those from `p.director.endpoints` with `refresh` method.